src/Microsoft.CSharp/tests/CSharpGetIndexBinderTests.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using Xunit;
+
+namespace Microsoft.CSharp.RuntimeBinder.Tests
+{
+    public class CSharpGetIndexBinderTests
+    {
+        [Fact]
+        public void ArrayIndexing()
+        {
+            dynamic d = new[] { 0, 1, 2, 3 };
+            for (int i = 0; i != 4; ++i)
+            {
+                Assert.Equal(i, d[i]);
+            }
+        }
```

Comment:
As per #13985 where I realised the tests I'm adding for System.Dynamic can serve as tests for Microsoft.CSharp.RuntimeBinder and put copies of them there, so as I complete those System.Dynamic tests I'm adding those that can serve here into this test project too.